### PR TITLE
status_server: Support online config update for configuration hosting platform

### LINKED ISSUE #11692: Support online config update for configuration hosting platform
## Development Task

Cloud platforms such as K8s will perform configuration management, and TiKV does not have write permissions on these platforms. Therefore, an interface needs to be added to support this situation.

### PR DESCRIPTION
**Pull request by ethercflow**
… platform (#11692)

close #11692

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### What is changed and how it works?

Close #11692  

What's Changed:  
- add a new API in status_server

### Related changes

- None

### Check List 

Tests 

- Manual test 
   1. Setup tikv-server
   2. run curl -X PUT

Side effects

No

### Release note 

```release-note
None.
```

### CODE REVIEW

**Comment by ethercflow** [comment]
@5kbpers @NingLin-P PTAL, thanks!

File: src/config.rs [hunk 1]
```diff
@@ -3231,6 +3231,21 @@ impl ConfigController {
...
+    }
+
+    pub fn update_from_toml_file(&self) -> CfgResult<()> {
+        let current = self.get_current();
+        let incoming = TiKvConfig::from_file(Path::new(&current.cfg_path), None);
+        let diff = current.diff(&incoming);
+        self._update(diff, None)
+    }
+
+    fn _update(
```
Comment (5kbpers) [inline src/config.rs:3244]: Prefer to name it `update_impl`. I think underscore prefixed function names is not commonly used in Rust

Comment (ethercflow) [reply src/config.rs]: Fixed, thanks!

**Review by 5kbpers — APPROVED** [review]
LGTM

File: src/config.rs [hunk 2]
```diff
@@ -3231,6 +3231,21 @@ impl ConfigController {
 
     pub fn update(&self, change: HashMap<String, String>) -> CfgResult<()> {
         let diff = to_config_change(change.clone())?;
+        self.update_impl(diff, Some(change))
+    }
+
+    pub fn update_from_toml_file(&self) -> CfgResult<()> {
```
Comment (NingLin-P) [inline src/config.rs:3237]: Plz add some test cases for this method

File: src/config.rs [hunk 3]
```diff
@@ -3231,6 +3231,21 @@ impl ConfigController {
...
+        let current = self.get_current();
+        let incoming = TiKvConfig::from_file(Path::new(&current.cfg_path), None);
```
Comment (NingLin-P) [inline src/config.rs:3239]: `TiKvConfig::from_file` will panic if there is no such file, better to change it to return an `Err` and we print a log here instead.

Comment (ethercflow) [reply src/config.rs]: Fixed, thanks!

File: src/config.rs (hunk 2 above)
Comment (ethercflow) [reply src/config.rs:3248]: Added, thanks!

**Review by NingLin-P — APPROVED** [review]
LGTM

**Pull request merged**